pbs_light.go

```diff
@@ -198,61 +200,86 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 
 	ch := make(chan bidResult)
 	sentBids := 0
-	for _, bidder := range pbs_req.Bidders {
-		if ex, ok := exchanges[bidder.BidderCode]; ok {
-			ametrics := adapterMetrics[bidder.BidderCode]
-			ametrics.RequestMeter.Mark(1)
-			if pbs_req.App == nil && pbs_req.GetUserID(ex.FamilyName()) == "" {
-				bidder.NoCookie = true
-				bidder.UsersyncInfo = ex.GetUsersyncInfo()
-				ametrics.NoCookieMeter.Mark(1)
-				continue
-			}
-			sentBids++
-			go func(bidder *pbs.PBSBidder) {
-				start := time.Now()
-				bid_list, err := ex.Call(ctx, pbs_req, bidder)
-				bidder.ResponseTime = int(time.Since(start) / time.Millisecond)
-				ametrics.RequestTimer.UpdateSince(start)
-				if err != nil {
-					switch err {
-					case context.DeadlineExceeded:
-						ametrics.TimeoutMeter.Mark(1)
-						bidder.Error = "Timed out"
-					case context.Canceled:
-						fallthrough
-					default:
-						ametrics.ErrorMeter.Mark(1)
-						bidder.Error = err.Error()
-					}
-				} else if bid_list != nil {
-					bidder.NumBids = len(bid_list)
-					am.BidsReceivedMeter.Mark(int64(bidder.NumBids))
-					for _, bid := range bid_list {
-						ametrics.PriceHistogram.Update(int64(bid.Price * 1000))
-						am.PriceHistogram.Update(int64(bid.Price * 1000))
-					}
-				} else {
-					bidder.NoBid = true
-					ametrics.NoBidMeter.Mark(1)
+	returnedBids := 0
+
+	// maxTime is the maximum amount of time to wait for the bidders to return responses back
+	maxTime := time.NewTimer(time.Millisecond * time.Duration(defaultTimeoutMS))
+
+	// process all of the bidders inside of a goroutine.
+	// we'll wait for the responses in the for loop below
+	go func(pbs_req *pbs.PBSRequest) {
```

Comment:
What benefit does this have?   Adding an extra goroutine doesn't guarantee concurrency here, I think it will behave very similar to the existing behavior but now it has the added overhead of a new goroutine.